user_sync/config.py

```diff
@@ -694,6 +611,138 @@ def describe_unused_values(self):
             messages.append("Found unused keys: %s in: %s" % (unused_keys, self.get_full_scope()))
         return messages
     
+class ConfigFileLoader(object):
+    @staticmethod
+    def load_from_yaml(filename, path_keys):
+        '''
+        loads a yaml file, processes the resulting dict to adapt values for keys
+        (the path to which is defined in PATH_KEYS) to a value that represents
+        a file reference relative to provided source file, and returns the
+        processed dict.
+        :type filename: str
+        :rtype dict
+        '''
+        try:
+            input_file = open(filename, 'r', 1)
```

Comment:
this doesn't do a close after the read.  the right way to do this is:
```python
with open(filename, 'r', 1) as input_file:
    yml = yaml.load(input_file)
```
That way the file is guaranteed to be closed on either success or error.